v2/tools/asoctl/internal/crd/cleaner.go

```diff
@@ -170,7 +162,33 @@ func (c *Cleaner) migrateObjects(ctx context.Context, objectsToMigrate *unstruct
 			continue
 		}
 
-		err := retry.OnError(c.migrationBackoff, isErrorFatal, func() error { return c.client.Update(ctx, &obj) })
+		originalVersionFieldPath := []string{"spec", "originalVersion"}
+
+		originalVersion, found, err := unstructured.NestedString(obj.Object, originalVersionFieldPath...)
+		if err != nil {
+			return errors.Wrap(err,
+				fmt.Sprintf("migrating %q of kind %s", obj.GetName(), obj.GroupVersionKind().Kind))
+		}
+
+		// If we don't find the originalVersion, it may not have been set.
+		// This can happen for some resources such as ResourceGroup which were handcrafted in versions prior to v2.0.0 and thus didn't have a StorageVersion.
+		if !found {
+			c.log.Info(
+				"originalVersion not found. Continuing with the latest.",
+				"name", obj.GetName(),
+				"kind", obj.GroupVersionKind().Kind)
+			// Here we set the originalVersion as active version since we can't find it
+			originalVersion = getVersionFromStoredVersion(activeVersion)
+		}
+
+		originalVersion = strings.Replace(originalVersion, "v1alpha1api", "v1beta", 1)
+		err = unstructured.SetNestedField(obj.Object, originalVersion, originalVersionFieldPath...)
```

Comment:
Should you only do this if `found`? Since if there wasnt' one to begin with you can't set it?